homeassistant/components/signal_messenger/notify.py

```diff
@@ -71,9 +72,21 @@ def send_message(self, message="", **kwargs):
                     filenames = [data[ATTR_FILENAME]]
                 else:
                     filenames.append(data[ATTR_FILENAME])
+            if ATTR_URLS in data:
+                urls = data[ATTR_URLS]
+                for url in urls:
+                    try:
+                        attachments_as_bytes.append(
+                            (requests.get(url, verify=False, timeout=10)).content
+                        )
+                    except Exception as ex:
```

Comment:
Feel like there should be some size limitation here? This could take down the server if you ask it to send a very large file.

Reply:
Added size limits checks and tests (d8e59e9b32d64a881b57421a921f317dd169e049)